Cargo.toml

```diff
@@ -42,14 +42,14 @@ sparkle = { version = "0.1", optional = true }
 osmesa-sys = { version = "0.1", optional = true }
 rwh_05 = { package = "raw-window-handle", version = "0.5.2", features = ["std"], optional = true }
 rwh_06 = { package = "raw-window-handle", version = "0.6.2", features = ["std"], optional = true }
+serial_test = "3.1.0"
```

Comment:
Why does serial_test need to be added to the main packages dependencies?

Reply:
It is to allow the compilation of android_example. It imports the test module from the main library in `android-example/rust/src/lib.rs:17` for implementation of JNI methods.

Reply:
If `serial_test` is required by the example, then be added to the Cargo.toml of the example crate? 
Adding a mod via `path = ` shouldn't be affected by the Cargo.toml of the foreign crate.

Reply:
android-example doesn't use serial_test directly. It [uses](https://github.com/servo/surfman/blob/b6a2125cee6fda6259e8240d4338b0c55a357a0d/android-example/rust/src/lib.rs#L17) the `test` module of the surfman crate. It is the test module in surfman that uses the serial_test, so I don't think adding it to android-example's manifest will work.

However, I think another approach to eliminate this dependency would be to conditionally import serial_test in surfman/test.rs using `cfg(not(feature = "sm-test"))` and similarly use `cfg_attr(not(feature="sm-test"), serial)` on all the test methods.

Reply:
I see. I believe we should at the very least make the `serial_test` dependency optional and move it behind a feature flag, since there is no need to pull in `serial_test` when compiling servo.

Reply:
In [5ae1ac4](https://github.com/servo/surfman/pull/303/commits/5ae1ac49f6e0f282f81a467f181e894214785b78), I've made the use of `#[serial]` conditional on `not(feature = "sm-test")` and I've move serial_test back to dev-dependencies.